fedora_messaging/_session.py

```diff
@@ -21,25 +21,88 @@
 import logging
 import signal
 import uuid
+import ssl
 
 from pika import exceptions as pika_errs
 import pika
 import jsonschema
+try:
+    # Versions of pika greater than the 0.11 series uses SSLOptions to configure
+    # a TLS connection.
+    from pika import SSLOptions
+except ImportError:
+    SSLOptions = None
 
 from . import config
 from .message import _schema_name, get_class, Message
 from .exceptions import (
-    Nack, Drop, HaltConsumer, ValidationError, PublishReturned, ConnectionException)
+    Nack, Drop, HaltConsumer, ValidationError, PublishReturned, ConnectionException,
+    ConfigurationException)
 
 _log = logging.getLogger(__name__)
 
 
+def _configure_tls_parameters(parameters):
+    """
+    Configure the pika connection parameters for TLS based on the configuration.
+
+    This modifies the object provided to it. This accounts for whether or not
+    the new API based on the standard library's SSLContext is available for
+    pika.
+
+    Args:
+        parameters (pika.ConnectionParameters): The connection parameters to apply
+            TLS connection settings to.
+    """
+    cert = config.conf['tls']['certfile']
+    key = config.conf['tls']['keyfile']
+    if cert and key:
+        _log.info('Authenticating with server using x509 (certfile: %s, keyfile: %s)',
+                  cert, key)
+    else:
+        cert, key = None, None
+
+    if SSLOptions is None:
+        parameters.ssl = True
+        parameters.ssl_options = {
+            'keyfile': key,
+            'certfile': cert,
+            'ca_certs': config.conf['tls']['ca_cert'],
+            'cert_reqs': ssl.CERT_REQUIRED,
+            'ssl_version': ssl.PROTOCOL_TLSv1_2,
+        }
+    else:
+        ssl_context = ssl.create_default_context()
+        try:
+            ssl_context.load_verify_locations(cafile=config.conf['tls']['ca_cert'])
+        except ssl.SSLError as e:
+            raise ConfigurationException(
+                'The "ca_cert" setting in the "tls" section is invalid: ' + str(e))
```

Comment:
Using `+` to concatenate strings is... unusual :-) People use the `.format()` string method these days ;-)

Reply:
I figured since it was at the end of the string it was shorter (LOC-wise) to just concatenate, but I'm happy to switch it to format

Reply:
As you wish, but `.format()` opens a parenthesis which lets you indent nicely on the new line ;-)